### PR TITLE
Fix `TableMixin` region only (no useful `colorColumn`)

### PR DESCRIPTION
**Pull request by nf-s**
### Fix `TableMixin` region only (no useful `colorColumn`)

There was an issue where CSVs which only had region mapping would show no data - even if `colorColumn` is set to a region column.

```json
{
      "type": "csv",
      "name": "region color test",
      "defaultStyle": {
        "color": {
          "colorColumn": "LGA_code",
          "regionColor": "orange"
        }
      },
      "url": "dcp-lgas.csv"
    },
```

### Checklist

-   [ ] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
-   [x] I've updated CHANGES.md with what I changed.

### CODE REVIEW

**Review by tephenavies — APPROVED** [review]

**Review by tephenavies — CHANGES_REQUESTED** [review]
This still doesn't act as before. Try https://data.gov.au/data/dataset/76327768-7e34-4464-b4a2-6bba32876d32/resource/de4745ba-44b5-496a-a1ca-ded413deb508/download/dcp-lgas.csv on http://ci.terria.io/fix-region-csv/. You should only get the regions that are in the dataset, not all regions coloured.

File: lib/Table/TableStyle.ts [hunk 1]
```diff
@@ -471,29 +468,52 @@ export default class TableStyle {
...
+        title: this.tableModel.name,
+        nullColor
+      });
     }
   }
 
+  @computed get defaultColor() {
+    return Color.fromCssColorString(defaultColor);
+  }
+
```
Comment (tephenavies) [inline lib/Table/TableStyle.ts]: This shouldn't be a computed because it doesn't use anything observable. Does it even need to be in this class?

Comment (nf-s) [reply lib/Table/TableStyle.ts]: Sorry I forgot to remove that - done!

File: lib/Table/TableStyle.ts [hunk 2]
```diff
@@ -471,19 +468,35 @@ export default class TableStyle {
...
+        color = this.regionColor;
+      } else if (colorTraits.nullColor) {
         color = Color.fromCssColorString(colorTraits.nullColor);
-      } else if (this.binColors.length > 0) {
-        color = this.binColors[0];
+      } else if (colorTraits.binColors && colorTraits.binColors.length > 0) {
+        color = Color.fromCssColorString(colorTraits.binColors[0]);
       } else if (colorId) {
         color = Color.fromCssColorString(getColorForId(colorId));
       }
```
Comment (nf-s) [inline lib/Table/TableStyle.ts:484]: I changed this.binColors to `colorTraits.binColors` as I think we should only use binColors here if they are defined manually (not the default binColors).

This means that `Color.fromCssColorString(getColorForId(colorId))` is used more - which makes it easier to see different datasets

**Comment by tephenavies** [comment]
It still doesn't behave well when adding https://data.gov.au/data/dataset/76327768-7e34-4464-b4a2-6bba32876d32/resource/de4745ba-44b5-496a-a1ca-ded413deb508/download/dcp-lgas.csv via web data. I'd expect only the regions in the file to be coloured (either in 1 colour, or in many colours I don't really mind)

**Comment by nf-s** [comment]
> 
> 
> It still doesn't behave well when adding https://data.gov.au/data/dataset/76327768-7e34-4464-b4a2-6bba32876d32/resource/de4745ba-44b5-496a-a1ca-ded413deb508/download/dcp-lgas.csv via web data. I'd expect only the regions in the file to be coloured (either in 1 colour, or in many colours I don't really mind)

Fixed

**Comment by tephenavies** [comment]
Thanks for fixing this up. It works with my examples now.

**Review by tephenavies — APPROVED** [review]

**Pull request merged**